### ISSUE TITLE
[🐞] Warned useOnXXX even for not empty components.

### ISSUE THREAD

**Issue by genki** [BuilderIO/qwik#2980]
### Which component is affected?

Qwik Runtime

### Describe the bug

When using `useOnXXX` in the component that including components like this:
```ts
export default component$(() => {
  useOnDocument("DOMContentLoaded", $(() => {
    ...
  }));
  return (
    <>
    <Link href="/">Test</Link>
    </>
  );
});
```

 it is warned
```
QWIK WARN Component has listeners attached, but it does not render any elements, injecting a new <script> element to attach listeners.
```

I think this warning is a bit surplus.

### Reproduction

https://stackblitz.com/edit/qwik-starter-uhcukh?file=src/routes/index.tsx

### Steps to reproduce

Please see the reproduction link.

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M2
    Memory: 78.86 MB / 24.00 GB
    Shell: 3.5.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 19.5.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.3.1 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Edge: 110.0.1587.41
    Firefox: 109.0.1
    Safari: 16.2
  npmPackages:
    @builder.io/qwik: file:../clone/qwik/packages/qwik => 0.17.6 
    @builder.io/qwik-city: file:../clone/qwik/packages/qwik-city => 0.1.1 
    undici: 5.14.0 => 5.14.0
```

**Comment by genki** [comment]
@zanettin 
I expect the warning is not necessary because the `<script>` tag insertion works.
I think it is not so special to make intermediate components consist of only child components without normal HTML tags.

**Comment by zanettin** [comment]
Hi @genki 
Thanks a lot for your reply 🙏 

> I expect the warning is not necessary because the <script> tag insertion works.

Imo this just works, because you use global events to attach a listener to. Let's check another example:

```
export default component$(() => {
  useOnDocument(
    "click",
    $(() => {
      console.log("===========> and action");
    })
  );

  return (
    <>
      <Link href="/">Test</Link>
    </>
  );
});
```

this would generate this script tag (which would not log anything)
```
<script type="placeholder" hidden q:id="z" on-document:click="/src/random_component_useondocument_ekuykhfkzcq.js#random_component_useOnDocument_EKuYKHfkZCQ"></script>
```

PS: Pls don't ask me what this script does in that case 😃 

Same would be if you try to attach anything using just `useOn`. From a technical perspective i am not sure how i would tackle that in plain JSX as i've mentioned above. There is just nothing i would be able to register a listener on.

Since i am now out of my knowledge zone, i have to hand this issue over to anyone from the core team with deeper knowledge about this. Sorry that i wasn't able to help you

**Comment by genki** [comment]
Thank you for explanation :)
I have made another example that uses the `click` event.
https://stackblitz.com/edit/qwik-starter-cefpjz?file=src/routes/index.tsx

This works fine, but about `useOn` instead of `useOnDocument` it doesn't work.
If there is a way to delegate the event handler to the child components, I think even the `useOn` possibly works as well.
Because there's something can be the event target in descendant components.

Let the issue revised,
* The warning on `useOnDocument` and `useOnWindow` may be surplus.
* The warning on `useOn` is currently correct. But it is going to be possible to omit if there would be the mechanism to delegate the event handler to the first element in descendant components.

**Comment by genki** [comment]
Fixed by #3031
